batavia/types/Tuple.js

```diff
@@ -143,19 +143,70 @@ batavia.types.Tuple = function() {
     };
 
     Tuple.prototype.__mod__ = function(other) {
-        throw new batavia.builtins.NotImplementedError("Tuple.__mod__ has not been implemented");
+        throw new batavia.builtins.TypeError("unsupported operand type(s) for %: 'tuple' and '" + batavia.type_name(other) + "'");
     };
 
     Tuple.prototype.__add__ = function(other) {
-        throw new batavia.builtins.NotImplementedError("Tuple.__add__ has not been implemented");
+		if (!batavia.isinstance(other, batavia.types.Tuple)) {
+			throw new batavia.builtins.TypeError('can only concatenate tuple (not "' + batavia.type_name(other) + '") to tuple')
```

Comment:
That seems to be a rendering bug in GitHub. I don't have that indentation in my source file.